### DISCUSSION TITLE
CO2 Property Discrepancy

### DISCUSSION THREAD

**Discussion by NCJazzMan65** [IDAES/idaes-pse#660] [General]
I have a flowsheet which models a supercritical CO2 cycle.  I'm using the Span-Wagner CO2 property package.  The flowsheet converges to a solution but when I inspect the results they are not what I expected.  I spot checked the state of some streams using the REFPROP add-in for Excel.  The enthalpy being calculated by IDAES is way off the enthalpy being calculated by REFPROP.  As an example, I have one stream which is at 0 C (273.15 K) and 6.4e6 Pa.  IDAES calculates an enthalpy of -13613 J/mol but REFPROP calculates 8668 J/mol for the same conditions.  I've double checked everything in my IDAES model and even looked at the Span-Wagner property package to confirm it applies to my range of conditions.  I cannot figure out why my IDAES model is giving me incorrect results even though it converged successfully (I understand converging and converging to the "right" solution are two different things).

Can anybody help?

**Comment by NCJazzMan65** [comment]
I should further clarify and say I'm using the htpx() method of the swco2 property package to calculate the enthalpy of specific streams in my spot check.

**Reply by andrewlee94** [reply]
My first comment is to recall that enthalpy is a relative quantity, and that is is not unexpected for the absolute value of enthalpy to vary significantly between different packages. The more important thing is to check whether the difference in enthalpy between two points (often the current state and a reference state of your choosing) is the same.

**Comment by NCJazzMan65** [comment]
Andrew...Thank you for your feedback.  I thought about that after I submitted my original post (having a senior moment today).  I did spot check the change in enthalpy across a couple of my unit models and there is still a discrepancy.  My flowsheet is somewhat complex and involves recycle streams.  I had to supply guesses for two tear streams.  The flowsheet converged but I'm not confident it converged to the "right" solution.  I'm going to simplify my flowsheet down to just a couple unit models to start and see if I get better agreement.

BTW...what are the reference conditions for the swco2 property package?

**Reply by andrewlee94** [reply]
@eslickj is the one who is best to ask about details of the SWCO2 package. If the differential enthalpy is still off then there is something that needs to be looked into somewhere.

**Comment by eslickj** [comment]
There could be a bug in there.  I tried to test this pretty extensively, but it's possible there are conditions where things go wrong.  I also left out the nonanalytic terms, so there may be a little more error close to the critical point.  If you find anything that doesn't look right I'll check it out. 

I implemented the Span-Wagner from : Journal of Physical and Chemical Reference Data 25, 1509 (1996); https://
doi.org/10.1063/1.555991.

The reference state is:
T = 298.15 K 
P = 0.101 325 MPa 
Reference enthalpy in the ideal gas state at Tref is 0 
Reference entropy in the ideal gas state at Tref, Pref:  0

**Comment by NCJazzMan65** [comment]
Update...everything appears to be good with the CO2 property package.  I had some other issues in my full model which were causing the apparent discrepancies.  I fixed those issues and now everything checks out.
